packages/configure-splash-screen/src/validators/FromJsonValidator.ts

```diff
@@ -0,0 +1,92 @@
+import { get, set } from 'lodash';
+import { DeepRequired, IsNever, KeyofObject, OptionalPromise, JsonShape } from '../types';
+
+/**
+ * This class is responsible for validating configuration object in a form of json and produce validated object based on validating `rules` added via `addRule` method.
+ */
+export default class FromJsonValidator<From extends JsonShape<To>, To extends object> {
```

Comment:
Congratulations on implementing an object validator! I too am curious whether you considered other NPM packages. As I mentioned on one of the calls I've had pretty good experience with [`zod`](https://github.com/vriad/zod) which, I guess, does the same thing (parses an unknown object into an object of a known TS type, inferring and parsing types and optionally validates)?

While I admire the heart you've put into this and stuff you must have learned along the way I'm a little afraid maintaining this may be a burden no one will want to take a couple of months from now and this will become a zombie file _here be dragons_.

Reply:
Well, it was my little experiment 😉 
I really like the `zod` idea and I'm willing to add it to our codebase somewhere. I'm not sure though if `zod` can provide validation & transformation at the same time, please take a look at following lines:
https://github.com/expo/expo-cli/pull/2297/files#diff-63c94c9e1d2966b124e8333ec934d795R55
https://github.com/expo/expo-cli/pull/2297/files#diff-63c94c9e1d2966b124e8333ec934d795R29
These represent the same think, but one in JSON and other in it's semantically-validated equivalent (`Color` is a tuple). What I'm curious about is: whether `zod` can convert for example string into some complex object/structure? 🤔 Nonetheless, IMO it's still worth to be added 👍